src/graph/impl/KokkosGraph_Distance2MIS_impl.hpp

```diff
@@ -854,128 +915,356 @@ struct D2_MIS_FixedPriority {
 
 template <typename device_t, typename rowmap_t, typename entries_t,
           typename labels_t>
-struct D2_MIS_Coarsening {
+struct D2_MIS_Aggregation {
```

Comment:
Workflow question: Who is responsible for allocating and deallocating the aggregate graph and when? My guess the aggregate graph has to be in scope longer than KK structs, correct?

Reply:
This struct just contains the implementation and goes out of scope when ``graph_mis2_aggregate()`` or ``graph_mis2_coarsen()`` return. But the actual coarsened graph (if you call e.g. ``graph_mis2_aggregate()`` followed by ``graph_explicit_coarsen()``)  just consists of rowptrs/entries views that go out of scope whenever you're done with them. The input (fine) graph and coarsened graph don't depend on each other.